storage.go

```diff
@@ -207,6 +236,26 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 	if opt.HasStorageClass {
 		putOptions.XCosStorageClass = opt.StorageClass
 	}
+	// SSE-C
+	if opt.HasServerSideEncryptionCustomerAlgorithm {
+		putOptions.XCosSSECustomerAglo, putOptions.XCosSSECustomerKey, putOptions.XCosSSECustomerKeyMD5, err = calculateEncryptionHeaders(opt.ServerSideEncryptionCustomerAlgorithm, opt.ServerSideEncryptionCustomerKey)
+		if err != nil {
+			return
+		}
+	}
+	// SSE-COS or SSE-KMS
+	if opt.HasServerSideEncryption {
+		putOptions.XCosServerSideEncryption = opt.ServerSideEncryption
+		if putOptions.XCosServerSideEncryption == ServerSideEncryptionCosKms {
+			putOptions.XOptionHeader = &http.Header{}
+			if opt.HasServerSideEncryptionCosKmsKeyID {
+				putOptions.XOptionHeader.Set("x-cos-server-side-encryption-cos-kms-key-id", opt.ServerSideEncryptionCosKmsKeyID)
```

Comment:
How about extracting them as const?

And let's leave a `FIXME` here as we can remove the usage of `XOptionHeader` when cos' SDK supports  `ServerSideEncryptionCosKmsKeyID`.